spec/takeaway_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'takeaway'
+
+describe Takeaway do
+  let(:menu) { double(:menu, dishes: [
+    { "Margherita" => 7.5 },
+    { "Bufala" => 8 },
+    { "Anchovy" => 8 },
+    { "Pepperoni" => 8.5 },
+    { "Calabria" => 8.5 },
+    { "Quattro Formaggi" => 10 },
+    { "Truffle" => 12.5 },
+    { "Gold Pizza" => 1200 }
+  ])
+  }
+  let(:menu_class) { double(:menu_class, new: menu) }
+  let(:text) { double(:text, send: nil) }
+  let(:text_class) { double(:text_class, new: text) }
+  let(:basket) { double(:basket, items: [{ name: "Margherita", quantity: 2, cost: 17.0 }]) }
+  let(:basket_class) { double(:basket_class, new: basket) }
```

Comment:
Lots of complexity in the doubling here - might suggest that there's an opportunity to simplify the design of the application by changing the way the classes interact. Something to think about :)